sdk/eventhub/event-hubs/test/node/disconnects.spec.ts

```diff
@@ -50,30 +50,86 @@ describe("disconnected", function() {
+       * would be invoked with 0 events and ignoring the `maxWaitTimeInSeconds` after a `disconnected` event.
+       *
+       * For a single `subscribe()` call, this test does the following:
+       * 1. Ensure events can be received normally before the `disconnected` event.
+       * 2. Ensure that the `maxWaitTimeInSeconds` is honoured after a `disconnected` event.
+       * 3. Ensure that events can be received normally after the `disconnected` event.
+       */
+      const consumer = new EventHubConsumerClient(
         EventHubConsumerClient.defaultConsumerGroupName,
         service.connectionString,
         service.path
       );
+
+      const producer = new EventHubProducerClient(service.connectionString, service.path);
+      const event1 = { body: "the first event" };
+      const event2 = { body: "the second event" };
+
+      const maxWaitTimeInSeconds = 10;
       const partitionId = "0";
-      const partitionProperties = await client.getPartitionProperties(partitionId);
-      const clientConnectionContext = client["_context"];
+      const partitionProperties = await consumer.getPartitionProperties(partitionId);
+      const clientConnectionContext = consumer["_context"];
+
+      // Send the first event after getting partition properties so that we can expect to receive it.
+      await producer.sendBatch([event1], { partitionId });
 
       let subscription: Subscription | undefined;
       let originalConnectionId: string;
 
+      let processEventsInvocationCount = 0;
+      let firstInvocationEndTime = 0;
       await new Promise((resolve, reject) => {
-        subscription = client.subscribe(
+        subscription = consumer.subscribe(
           partitionId,
           {
             processEvents: async (data) => {
+              processEventsInvocationCount++;
               should.exist(data);
-              should.equal(data.length, 0);
-              if (!originalConnectionId) {
+              if (processEventsInvocationCount === 1) {
+                // 1. Ensure events can be received normally before the `disconnected` event.
+                should.equal(
+                  data.length,
+                  1,
+                  "Expected to receive 1 event in first processEvents invocation."
+                );
+                should.equal(data[0].body, event1.body);
                 originalConnectionId = clientConnectionContext.connectionId;
                 // Trigger a disconnect on the underlying connection.
                 clientConnectionContext.connection["_connection"].idle();
-              } else {
+                firstInvocationEndTime = Date.now();
+              } else if (processEventsInvocationCount === 2) {
+                // 2. Ensure that the `maxWaitTimeInSeconds` is honoured after a `disconnected` event.
+                // No new events should have been received at this point since we received the last event in the previous invocation.
```

Comment:
processEvents is invoked even if there is no event received?
(That's interesting, and is different from service-bus)

Reply:
Yes. Note that we do send an array of events (even empty) _always_ though whereas in ServiceBus we'd be delivering an null/undefined message.

They are still inconsistent on this, however.

Reply:
Yes...we've thought about adding an option so that it doesn't get invoked if there are no events but the original reason for having it was as a 'heartbeat' so you knew that the subscription was working even if no events were seen for a long time.